specs/separate_configuration_files_per_component.adoc

```diff
@@ -0,0 +1,232 @@
+import flow the configuration file gets appended with SDS specific
+configurations while installation of SDS integration bits on the storage nodes.
+To avoid this we have to store the configuration details into layer wise.
+
+1. Store the configuration details into different files as per modules, And
+   configure modules using those files. (high priority).
+2. Each modules should store its own configuration details to etcd and configured
+   via api request (Less priority).
+
+Only node_agent configuration file have its own configuration details and
+etcd configuration details.
+
+At the time of installation, Each module's configuration details should store in
+different files with proper name. Always tendrl modules should configured from
+those files only.
+When module run the configuration details should be stored in etcd.
+
+* Integration modules should store its configuration details under cluster
+  specific hierarchy in the central store.
+* Other modules should store its configuration with proper hierarchy.
+
+The configuration in the central store needs to be treated as an object
+hierarchy, which appropriate atoms and flows to allow modification of this
+object hierarchy. If any modification on configurations in etcd, Then it should
+affect a local configuration files also. But any changes in local configuration
+file should not affect configurations in etcd. Any changes in integration
+specific configuration in etcd will affect config files of particular cluster
+nodes only. But changes in other configuration in etc will affect all nodes
+local config files.
+
+== Use Cases
+
+Centralized deployment-wide or cluster-wide configuration in the central store,
+which can be overridden per component via configuration files or command line
+options.
+
+== Proposed change
+
+* Modify the sample configuration files in each modules. It should contain
+  its own and relevant configuration information only.
+* Rename the sample configuration files in each modules with proper name.
+* Modify the spec file in each module to copy the configuration file with
+  proper name.
+* Create a new python file each module to keep configuration details.
+* Modify the config file in each module  to read the configuration details from
+  particular local config file.
+* Modify the manager in each module to read the configuration details from newly
+  created file and store it in in etcd. Integration specific modules should
+  store configuration details under cluster specific hierarchy in the central
+  store.
+* Modify the yaml file in each module with new object, atoms, flows, pre-run,
+  post-run, attributes, type, uuid, version, inputs, description for the
+  configuration.
+* Create atoms and flows to allow modification on configuration in etcd. It
+  should modify local configuration file also.
+* Modify tendrl definition file in node_agent to store node_agent specific
+  configuration details.
+
+== Alternatives
+
+None
+
+== Data model impact
```

Comment:
Refer https://github.com/Tendrl/specifications/pull/20/files#diff-a7f219a4f7014686f05ffba6d50dbdcdR250

We need similar Config object